File: _includes/code/howto/configure.backups.java
// How-to: Configure -> Backups - Java examples
// START CreateBackup
package io.weaviate;

import io.weaviate.client.Config;
import io.weaviate.client.WeaviateClient;
import io.weaviate.client.base.Result;
import io.weaviate.client.v1.backup.model.Backend;
import io.weaviate.client.v1.backup.model.BackupCreateResponse;

public class App {
  public static void main(String[] args) {
    Config config = new Config("http", "localhost:8080");
    WeaviateClient client = new WeaviateClient(config);

    Result<BackupCreateResponse> result = client.backup().creator()
      .withIncludeClassNames("Article", "Publication")
      .withBackend(Backend.FILESYSTEM)
      .withBackupId("my-very-first-backup")
      .withWaitForCompletion(true)
      .run();

    // END CreateBackup

    // START StatusCreateBackup
    Result<BackupCreateStatusResponse> result = client.backup().createStatusGetter()
      .withBackend(Backend.FILESYSTEM)
      .withBackupId("my-very-first-backup")
      .run();
    // END StatusCreateBackup

    // START RestoreBackup
    Result<BackupRestoreResponse> result = client.backup().restorer()
      .withExcludeClassNames("Article")
      .withBackend(Backend.FILESYSTEM)
      .withBackupId("my-very-first-backup")
      .withWaitForCompletion(true)
      .run();
    // END RestoreBackup

    // START StatusRestoreBackup
    Result<BackupRestoreStatusResponse> result = client.backup().restoreStatusGetter()
      .withBackend(Backend.FILESYSTEM)
      .withBackupId("my-very-first-backup")
      .run();
    // END StatusRestoreBackup

    // START CreateBackup
    if (result.hasErrors()) {
      System.out.println(result.getError());
      return;
    }
    System.out.println(result.getResult());
  }
}
// END CreateBackup

File: _includes/code/howto/configure.backups.go
// How-to: Configure -> Backups - Go examples
// START CreateBackup
package main

import (
	"context"
	"fmt"

	"github.com/weaviate/weaviate-go-client/v4/weaviate"
	"github.com/weaviate/weaviate-go-client/v4/weaviate/backup"
)

func main() {
	cfg := weaviate.Config{
		Host:   "localhost:8080",
		Scheme: "http",
	}
	client, err := weaviate.NewClient(cfg)
	if err != nil {
		panic(err)
	}

	result, err := client.Backup().Creator().
		WithIncludeClassNames("Article", "Publication").
		WithBackend(backup.BACKEND_FILESYSTEM).
		WithBackupID("my-very-first-backup").
		WithWaitForCompletion(true).
		WithCompressionConfig(backup.Compression{
			Level:         backup.BackupConfigCompressionLevelBestSpeed,
			ChunkSize:     512,
			CPUPercentage: 80,
		}).
		Do(context.Background())

	// END CreateBackup

	// START StatusCreateBackup
	result, err := client.Backup().CreateStatusGetter().
		WithBackend(backup.BACKEND_FILESYSTEM).
		WithBackupID("my-very-first-backup").
		Do(context.Background())
	// END StatusCreateBackup

	// START RestoreBackup
	result, err := client.Backup().Restorer().
		WithExcludeClassNames("Article").
		WithBackend(backup.BACKEND_FILESYSTEM).
		WithBackupID("my-very-first-backup").
		WithWaitForCompletion(true).
		WithCPUPercentage(80).
		Do(context.Background())
	// END RestoreBackup

	// START StatusRestoreBackup
	result, err := client.Backup().RestoreStatusGetter().
		WithBackend(backup.BACKEND_FILESYSTEM).
		WithBackupID("my-very-first-backup").
		Do(context.Background())
	// END StatusRestoreBackup

	// START CreateBackup
	if err != nil {
		panic(err)
	}
	fmt.Printf("%v", result)
}
// END CreateBackup
